File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/SurchargeSpecificOutputMapper.cs
using SurchargeSpecificOutputSdk = OnlinePayments.Sdk.Domain.SurchargeSpecificOutput;
using SurchargeSpecificOutputDto = Business.Domain.Payments.PaymentDetails.SurchargeSpecificOutput;

namespace Infrastructure.Mappers.PaymentDetails;

public static class SurchargeSpecificOutputMapper
{
    public static SurchargeSpecificOutputDto Map(SurchargeSpecificOutputSdk? response)
    {
        return new()
        {
            SurchargeRate = SurchargeRateMapper.Map(response?.SurchargeRate),
            SurchargeAmount = AmountOfMoneyMapper.Map(response?.SurchargeAmount),
            Mode = response?.Mode
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/SurchargeSpecificOutputMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\SurchargeSpecificOutput as SurchargeSpecificOutputDto;
use OnlinePayments\Sdk\Domain\SurchargeSpecificOutput as SurchargeSpecificOutputSdk;

final class SurchargeSpecificOutputMapper
{
    public static function mapFromSdkResponse(?SurchargeSpecificOutputSdk $response): ?SurchargeSpecificOutputDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new SurchargeSpecificOutputDto();
        $dto->surchargeRate = SurchargeRateMapper::mapFromSdkResponse($response?->surchargeRate ?? null);
        $dto->surchargeAmount = AmountOfMoneyMapper::mapFromSdkResponse($response?->surchargeAmount ?? null);
        $dto->mode = $response?->mode ?? null;

        return $dto;
    }
}
